compile error with ADIOS2 support on

**Describe the bug**

There is a compiling error when ADIOS2 support is switched on. This applies to both v4.0 and the latest master branch.

```
mpif90 -cpp -O3 -funroll-loops -floop-optimize -g -Warray-bounds -fcray-pointer -fbacktrace -ffree-line-length-none -fallow-argument-mismatch -DADIOS2 -I./src -I./decomp2d -cpp -O3 -funroll-loops -floop-optimize -g -Warray-bounds -fcray-pointer -fbacktrace -ffree-line-length-none -fallow-argument-mismatch -DDOUBLE_PREC -DVERSION=\"v4.0-249-g8d7b078\"   -DADIOS2_USE_MPI -I/path/to/adios2/include/adios2/fortran   -c decomp2d/io.f90
decomp2d/io.f90:139:87:

  139 |   call adios2_init(adios, trim(config_file), MPI_COMM_WORLD, adios2_debug_mode, ierror)
      |                                                                                       1
Error: There is no specific subroutine for the generic 'adios2_init' at (1)
make: *** [Makefile:110: decomp2d/io.o] Error 1
```

**To Reproduce**

make IO=adios2 ADIOS2DIR=/path/to/adios2 #with MPI wrapper in $PATH

--- a/decomp2d/io.f90
+++ b/decomp2d/io.f90
@@ -136,7 +136,7 @@ contains
   !! TODO: make this a runtime-option
   adios2_debug_mode = .true.
 
-  call adios2_init(adios, trim(config_file), MPI_COMM_WORLD, adios2_debug_mode, ierror)
+  call adios2_init(adios, trim(config_file), MPI_COMM_WORLD, ierror)
   if (ierror.ne.0) then
      print *, "Error initialising ADIOS2 - is adios2_config.xml present and valid?"
      call MPI_ABORT(MPI_COMM_WORLD, -1, ierror)

--- a/decomp2d/io.f90
+++ b/decomp2d/io.f90
@@ -128,14 +128,10 @@ contains
 
 #ifdef ADIOS2
   integer :: ierror
-  logical :: adios2_debug_mode
   character(len=80) :: config_file="adios2_config.xml"
 #endif
 
 #ifdef ADIOS2
-  !! TODO: make this a runtime-option
-  adios2_debug_mode = .true.
-
   call adios2_init(adios, trim(config_file), MPI_COMM_WORLD, ierror)
   if (ierror.ne.0) then
      print *, "Error initialising ADIOS2 - is adios2_config.xml present and valid?"

